src/wms/src/main/java/org/geoserver/wms/map/GetMapKvpRequestReader.java

```diff
@@ -325,51 +365,110 @@ public GetMapRequest read(Object request, Map kvp, Map rawKvp) throws Exception
 
             URL styleUrl = getMap.getStyleUrl();
 
-            if (getMap.getValidateSchema().booleanValue()) {
-                InputStream input = Requests.getInputStream(styleUrl);
-                List errors = null;
-
+            InputStream input = null;
+            if (styleUrl.getProtocol().toLowerCase().indexOf("http") == 0) {
```

Comment:
This block has become big enough that I'd suggest factoring it out in its own method if possible (e.g., getStyleStream(styleURL))